vyper/compiler/__init__.py

```diff
@@ -121,6 +128,13 @@ def compile_from_file_input(
         for output_format in output_formats:
             if output_format not in OUTPUT_FORMATS:
                 raise ValueError(f"Unsupported format type {repr(output_format)}")
+            elif (
+                file_input.resolved_path.suffix == ".vyi"
+                and output_format not in INTERFACE_OUTPUT_FORMATS
+            ):
```

Comment:
multiline expression. also, doesn't need to be an elif since line 130 fastpaths out.

Reply:
8048d4f19a2305152d22673f077e4b2ba6b36cfc